lib/logstash/util/compress.rb

```diff
@@ -0,0 +1,116 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/environment"
+
+module LogStash::Util
+  module Zip
+
+    extend self
+
+    def extract(source, target)
+      require "zip"
+      require "fileutils"
+
+      puts "Extracting #{source} to #{target}"
```

Comment:
I don't think these methods should `puts`, you cannot assume in what context they will be used: it could be used within logstash as some point where the `@logger` should be used for logging information. I think you should move all `puts` inside the plugin manager methods that uses these compress methods.